code/script.js

```diff
@@ -1,325 +1,824 @@
+
+    second++;
+
+    if (second == 60) {
+      minute++;
+      second = 0;
+    }
+    if (minute == 60) {
+      hour++;
+      minute = 0;
+    }
+  }, 1000);
+};
+/*********************************************************************/
+// For Modal Window
+
+// Here I am getting data from localStorage
+const getDataFromLocalStorage = () => {
+  if (localStorage.length === 1) {
+    modalDisplayArr = JSON.parse(localStorage.getItem("items"));
+  }
+};
+
+// checking the best scored game
+const checkBextGame = (arr) => {
+  arr.forEach((item) => {
+    if (item.howManyGuesses <= bestScore) {
+      bestScore = item.howManyGuesses;
+      bestPlayerObj = item;
+    }
+  });
+
+  // if there are the some score as the best score, then it filters and make it into a new array
+
+  const bestScoreArr = arr.filter((item) => item.howManyGuesses === bestScore);
+  const bestPlayerBox = document.createElement("div");
+
+  if (bestScoreArr.length < 5) {
+    bestScoreArr.forEach((item) => {
+      bestPlayerBox.classList.add("best-players-box");
+      bestPlayerBox.innerHTML += `
+    <p>Player: ${item.playerName}</p>
+    <p>Guesses: ${item.howManyGuesses}</p>
+    <p>Time: ${item.time}</p>
+    `;
+      bestScoreContainer.insertAdjacentElement("beforeend", bestPlayerBox);
+    });
+  } else {
+    const ajustedBestScoreArr = bestScoreArr.slice(0, 4);
+    ajustedBestScoreArr.forEach((item) => {
+      bestPlayerBox.classList.add("best-players-box");
+      bestPlayerBox.innerHTML += `
+    <p>Player: ${item.playerName}</p>
+    <p>Guesses: ${item.howManyGuesses}</p>
+    <p>Time: ${item.time}</p>
+    `;
+      bestScoreContainer.insertAdjacentElement("beforeend", bestPlayerBox);
+    });
+  }
+};
+// This creates table tr
+const createHTMLForTable = (arr) => {
+  // creating html for each item in an array
+  for (let i = 0; i < arr.length; i++) {
+    tbody.innerHTML += `<tr>
+  <td>${arr[i].playerName}</th>
+  <td>${arr[i].howManyGuesses}</th>
+  <td>${arr[i].time}</th>
+  <tr>
+`;
+    //  addting trElement the end of the childelement of Tbody
+    table.insertAdjacentElement("beforeend", tbody);
+  }
+  if (arr.length === 1) {
+    bestPlayerBox.insertAdjacentHTML("beforeend", "There is no comparable data");
+  }
+};
+
+// Create table element for a table on a modal window.
+const alartMessageBox = document.createElement("div");
+const createTableElement = (arr) => {
+  if (arr.length < 10) {
+    createHTMLForTable(arr);
+  } else {
+    const ajustedArr = arr.slice(0, 9);
+    createHTMLForTable(ajustedArr);
+
+    alartMessageBox.classList.add("table-alert-text");
+    alartMessageBox.textContent = "Please clear the data to store new data";
+    table.insertAdjacentElement("afterend", alartMessageBox);
+  }
+};
+
+/****************** Modal Event Handlers ************************************/
+
+// This is a button you can clean up localstorage and data that is displayed in a table.
+clearGameBtn.addEventListener("click", () => {
```

Comment:
should all players be able to clear the score board?

Reply:
I thought so as I am using localStorage here and wanted to create something to clean it up. And I didn't really think about it could be also nice to store all the data, and compare it. But for now, it is too tricky for me to dig deeper than this. But I don't really know how localStorage works. Is that a bad practice to store so much data there? How storing so much data could affect the performance... This is also a good point. Thank you for pointing out.